src/levanter/data/sharded_datasource.py

```diff
@@ -238,6 +243,11 @@ def open_shard_at_row(self, shard_name: str, row: int) -> Iterator[str]:
                     data = json.load(f)
                     for doc in data[row:]:
                         yield doc[self.text_key]
+                case ".parquet":
+                    table = pq.read_table(f)
```

Comment:
read_table is expensive in the general case and it would be better to look at the metadata to figure out which row group to start on and then use `read_row_group` I think

Reply:
Thanks for suggesting this, incorporated this in #766